app/services/idv/aamva_state_maintenance_window.rb

```diff
@@ -59,150 +65,141 @@ class AamvaStateMaintenanceWindow
+        Window.new(cron: '30 7 * * Sun#3', duration: 2.5.hours),
+        Window.new(cron: '30 7 * * Sun#4', duration: 2.5.hours),
       ],
       'SC' => [
         # Sunday 6:00 pm. to 10:00 pm. ET.
-        { cron: '0 18 * * Sun', duration_minutes: 4 * 60 },
+        Window.new(cron: '0 18 * * Sun', duration: 4.hours),
       ],
       'TN' => [
         # Last Sunday of every month from 11:00 pm Sunday to 2:00 am. Monday ET
-        { cron: '0 23 * * Sun#last', duration_minutes: 3 * 60 },
+        Window.new(cron: '0 23 * * Sun#last', duration: 3.hours),
       ],
       'TX' => [
         # Saturday 9:00 pm. to Sunday 7:00 am. ET.
-        { cron: '0 21 * * Sat', duration_minutes: 10 * 60 },
+        Window.new(cron: '0 21 * * Sat', duration: 10.hours),
       ],
       'UT' => [
         # 3rd Sunday of every month 1:00 am. to 9:00 am. ET
-        { cron: '0 1 0 0 Sun#3', duration_minutes: 8 * 60 },
+        Window.new(cron: '0 1 * * Sun#3', duration: 8.hours),
       ],
       'VA' => [
         # Daily 5:00 am. to 5:30 am. ET
-        { cron: '0 5 * * *', duration_minutes: 30 },
+        Window.new(cron: '0 5 * * *', duration: 30.minutes),
         # Sunday morning maintenance 3:00 am. to 5 am. ET.
-        { cron: '0 3 * * Sun', duration_minutes: 2 * 60 },
+        Window.new(cron: '0 3 * * Sun', duration: 2.hours),
         # "Might not respond for short spells, daily between 7 pm and 8:30 pm." (not modeling this)
       ],
       'VT' => [
         # Daily maintenance from midnight to 5 am. ET.
-        { cron: '0 0 * * *', duration_minutes: 5 * 60 },
+        Window.new(cron: '0 0 * * *', duration: 5.hours),
       ],
       'WA' => [
         # Maintenance from Saturday 9:45 pm. to Sunday 8:15 am. ET.
-        { cron: '45 21 * * Sat', duration_minutes: 10.5 * 60 },
+        Window.new(cron: '45 21 * * Sat', duration: 10.5.hours),
       ],
       'WI' => [
         # Downtime on Tuesday – Saturday typically between 3 – 4 am ET.
-        { cron: '0 3 * * Tue-Sat', duration_minutes: 60 },
+        Window.new(cron: '0 3 * * Tue-Sat', duration: 1.hour),
         # Downtime on Sunday from 6 – 10 am. ET.
-        { cron: '0 6 * * Sun', duration_minutes: 4 * 60 },
+        Window.new(cron: '0 6 * * Sun', duration: 4.hours),
       ],
       'WV' => [
         # Sunday 6:00 am. to 6:20 am. ET
-        { cron: '0 6 * * Sun', duration_minutes: 20 },
+        Window.new(cron: '0 6 * * Sun', duration: 20.minutes),
       ],
       'WY' => [
         # Daily, 2 am. to 5 am. ET.
-        { cron: '0 2 * * *', duration_minutes: 3 * 60 },
+        Window.new(cron: '0 2 * * *', duration: 3.hours),
       ],
     }.freeze
 
-    PARSED_MAINTENANCE_WINDOWS = MAINTENANCE_WINDOWS.transform_values do |windows|
-      Time.use_zone(TZ) do
-        windows.map do |window|
-          cron = Fugit.parse_cron(window[:cron])
-          { cron: cron, duration_minutes: window[:duration_minutes] }
-        end
-      end
-    end.freeze
```

Comment:
This clunkiness was for inline readability. Now that we have a Window Data object, though, I was able to implement a lil' `cron_val` method that performs this instead of building a whole separate data structure.

Academically, this lacks memoization since it's implemented on a frozen structure, which isn't great. IMHO this is acceptable because we're doing the `parse_cron` call after filtering down to the (US) state level, so worse case it's only a few calls, which seems to only take a millisecond or so in local testing.

Reply:
Very optional, but you might be able to throw a `lazy` into the `windows_for_state` return value to get some optimization back, so that we only map up to the first `.any?` hit.

https://ruby-doc.org/core-2.7.0/Enumerator/Lazy.html